From 3f1926b1283766a2c1aae6dbea383c2079e27d23 Mon Sep 17 00:00:00 2001
From: Eberhard Beilharz <eb1@sil.org>
Date: Fri, 16 Dec 2011 18:09:18 +0100
Subject: [PATCH] bug-novell-463149: Fix text position on buttons

Vertically centered text on buttons didn't show up centered. The
referenced bug reports it on Windows. Matters were worse on Linux.
This patch tries to fix the problem on both Windows and Linux. On Linux
the positioning is improved although probably still not perfect.

Change-Id: Iec641faddf5619d77b9c8437cf4dd046da57dc47
---
 .../System.Windows.Forms/TextRenderer.cs           |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

Index: fieldworks-mono/mcs/class/Managed.Windows.Forms/System.Windows.Forms/TextRenderer.cs
===================================================================
--- fieldworks-mono.orig/mcs/class/Managed.Windows.Forms/System.Windows.Forms/TextRenderer.cs
+++ fieldworks-mono/mcs/class/Managed.Windows.Forms/System.Windows.Forms/TextRenderer.cs
@@ -534,8 +534,8 @@
 			if ((flags & TextFormatFlags.WordEllipsis) == TextFormatFlags.WordEllipsis || (flags & TextFormatFlags.EndEllipsis) == TextFormatFlags.EndEllipsis || (flags & TextFormatFlags.WordBreak) == TextFormatFlags.WordBreak) {
 				r.Width -= 4;
 			}
-			if ((flags & TextFormatFlags.VerticalCenter) == TextFormatFlags.VerticalCenter) {
-				r.Y += 1;
+			if ((flags & TextFormatFlags.VerticalCenter) == TextFormatFlags.VerticalCenter && XplatUI.RunningOnUnix) {
+				r.Y -= 1;
 			}
 
 			return r;
